src/main/java/seedu/address/logic/parser/SessionCommandParser.java

```diff
@@ -0,0 +1,38 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SESSION;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.AttendanceCommand;
+import seedu.address.logic.commands.SessionCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Session;
+
+/**
+ * Parses input command and creates a new SessionCommand object.
+ */
+public class SessionCommandParser implements Parser<SessionCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the SessionCommand
+     * and returns a SessionCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    @Override
+    public SessionCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argumentMultimap = ArgumentTokenizer.tokenize(args, PREFIX_SESSION);
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(argumentMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AttendanceCommand.MESSAGE_USAGE), pe);
+        }
+        String session = argumentMultimap.getValue(PREFIX_SESSION).orElse("");
+        return new SessionCommand(index, new Session(session));
```

Comment:
You should probably use `ParserUtil#parseSession` here and then use that for the SessionCommand. This is how it's implemented in `AddCommandParser` too.

`AttendanceCommandParser` should probably have this too. Homework and GradeProgress don't need this now since they have no invalid input. You can run the app and enter an invalid attendance with your attendance command vs entering an invalid phone number in the add command to see the diff

Reply:
Hope i got what you intended to tell me right :-)